src/commands/deployAzurePipeline/configureCicdPipeline/configurePipeline.ts

```diff
@@ -0,0 +1,47 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import { resourceNode } from './models/resourceNode';
+const CONFIGURE_PIPELINE_COMMAND = 'configure-cicd-pipeline';
+const DEPLOY_TO_AZURE_EXTENSION_ID = 'ms-vscode-deploy-azure.azure-deploy';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void> {
+    const deployToAzureExtensionInstalled = isDeployToAzureExtensionInstalled();
+    if (deployToAzureExtensionInstalled) {
+        const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+        if (cloudExplorer.available) {
+            const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+            if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+                const cluster = { resource: clusterTarget.cloudResource, subscriptionId: clusterTarget.cloudResource.subscription.subscriptionId };
+                await executeDeployToAzureExtensionInstalled(CONFIGURE_PIPELINE_COMMAND, cluster);
+            } else {
+                vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+            }
+        }
+    } else {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
+        await vscode.commands.executeCommand('extension.open', DEPLOY_TO_AZURE_EXTENSION_ID);
+    }
+}
+
+function isDeployToAzureExtensionInstalled(): boolean {
+    const pipelinesExtension = vscode.extensions.getExtension(DEPLOY_TO_AZURE_EXTENSION_ID);
+    if (!pipelinesExtension) {
+        return false;
+    }
+    return true;
+}
+
+async function executeDeployToAzureExtensionInstalled(commandToRun: string, cluster: resourceNode): Promise<void> {
+    const listOfCommands = vscode.commands.getCommands(true);
+    if ((await (listOfCommands)).includes(commandToRun)) {
```

Comment:
💡 Please refer to my comments here: https://github.com/Azure/vscode-aks-tools/pull/42#discussion_r502048401 , Thank you.

Reply:
Cool, I see the return type is thenable <string[]> sire the you can await for the line above or whatever you find is best. Can you please double check. Thank you.

Reply:
cool, will update.